File: ICPC contest/Latin america/RPC/2021/01 LHCDP/n.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    int n;
    cin>>a>>n;
    while(a%2==0){
        a/=2;
        cont2++;
    }
    while(a%5 ==0){
        a/=5;
        cont5++;
    }
    if(a>1 && n!= 0){
        cout<<"Precision Error"<<endl;
    }
    else if(n==0){
        cout<<0<<" "<<1<<endl;
    }
    int v = cont2*n;
    int w = cont5*n;
    int m = max(v,w);
    
    cout<<m<<" "<<endl;

    return 0;
}

File: ICPC contest/Latin america/RPC/2021/01 LHCDP/n.py
import sys
for line in sys.stdin:
    a,n = line.split()
    a = int(a)
    n = int(n)
    cont2 = 0
    cont5 = 0
    while a%2==0:
        a = a/2
        cont2 = cont2+1
    while a%5 ==0:
        a = a/5
        cont5 = cont5+1
    if a>1 and n!= 0:
        print("Precision Error")
        continue
    if n==0:
        print("0 1")    
        continue
    v = cont2*n
    w = cont5*n
    m = max(v,w)
    print(m,(2**(m-v))*(5**(m-w)))